spec/namespaces/indices.yaml

```diff
@@ -1288,23 +1289,23 @@ paths:
           $ref: '#/components/responses/indices.forcemerge@200'
   /{index}/_mapping:
     get:
-      operationId: indices.get_mapping.1
-      x-operation-group: indices.get_mapping
+      operationId: indices.get_index_mapping.0
+      x-operation-group: indices.get_index_mapping
```

Comment:
Renaming this will cause issues for client generation, as these two are already treated as the same operation merely picking between the paths based on index being specified or not

Reply:
How should I solve this? The two APIs have effectively different parameters, as the `_mapping` API takes an index query parameter and the `{index}/_mapping` API does not. The linter complains that the two must have the same parameters.

Reply:
.NET's logic is that it unions all parameters from all variants and just prefers using the path over the query string if two parameters with the same name but different locations exist. Maybe we could loosen the linter slightly to just validate the variants have the same set of named parameters regardless of location?

Is there any reason to ever use this query string variant over the path variant?

Reply:
No reason, but the API supports it. We could of course not document it, but 🤷‍♂️ 

WDYT @nhtruong?

Reply:
@dblock Our clients support optional path parameters. Operations of the same group are only required to share an identical set of querystring params. The only requirement for path params is that if 2 operations support the same path params, then those params must be identical.

Reply:
The [Clients Generator Guide](https://github.com/opensearch-project/opensearch-api-specification/blob/main/CLIENT_GENERATOR_GUIDE.md) touched on this:
> If two operations have identical HTTP methods, but different paths: use the path that best matches the path parameters provided.

Reply:
> How should I solve this? The two APIs have effectively different parameters, as the _mapping API takes an index query parameter and the {index}/_mapping API does not. The linter complains that the two must have the same parameters.

The clients prioritize path parameters. If `index` is provided in the `clients.indices.get_mapping({ index: 'books' })` then `index` will extracted as a path param, and `/{index}/_mapping` will be used.

Reply:
Ended up just adding it to possible query parameters, thanks both!

Reply:
@Xtansia The clients gen will need to account for this situation when there are path and query params that share the same names. Especially for method documentation and typing for the `params` object. Should we also enforce that params of the same name, regardless of whether they are path or query, should have identical description and schema?